cdap-ui/app/features/hydrator/services/detail/stores/pipeline-detail-nodeconfig-store.js

```diff
@@ -37,7 +37,17 @@ angular.module(PKG.name + '.feature.hydrator')
     this.setState = function(plugin) {
       onPluginChange.call(this, plugin);
     };
+    this.setPlugin = function(plugin) {
+      this.plugin = plugin;
+      var type = this.isSource ? 'source': false;
```

Comment:
@anwar6953 I could not do this in this PR. I will make sure we do it in subsequent PRs. Right now merging this as @nitinmotgi is waiting on this change.